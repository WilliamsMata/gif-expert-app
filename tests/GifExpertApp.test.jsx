import { fireEvent, render, screen } from "@testing-library/react";
import { GifExpertApp } from "../src/GifExpertApp";

describe("Pruebas en <GifExpertApp />", () => {
  const inputValue = "Pokemon";

  test("Debe de hacer match con el Snapshot", () => {
    const { container } = render(<GifExpertApp />);

    expect(container).toMatchSnapshot();
  });

  test("Debe contener un H1", () => {
    render(<GifExpertApp />);

    expect(screen.findByRole("heading", { level: 1 })).toBeTruthy();
  });

  test("Hacer enviÃ³ al formulario debe agregar el valor del input", () => {
    render(<GifExpertApp />);
    const input = screen.getByRole("textbox");
    const form = screen.getByRole("form");

    fireEvent.input(input, { target: { value: inputValue } });
    fireEvent.submit(form);

    expect(screen.getByText(inputValue)).toBeTruthy();
  });

  test("Si el valor del input ya se encuentra no debe ser agregado nuevamente", () => {
    render(<GifExpertApp />);
    const input = screen.getByRole("textbox");
    const form = screen.getByRole("form");

    fireEvent.input(input, { target: { value: inputValue } });
    fireEvent.submit(form);
    fireEvent.input(input, { target: { value: inputValue } });
    fireEvent.submit(form);

    expect(screen.getAllByText(inputValue)).toHaveLength(1);
  });
});
