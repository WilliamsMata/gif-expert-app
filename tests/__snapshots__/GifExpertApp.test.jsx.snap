// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifExpertApp /> Debe de hacer match con el Snapshot 1`] = `
<div>
  <h1>
    GifExpertApp
  </h1>
  <form
    aria-label="form"
  >
    <input
      placeholder="Buscar GIF"
      type="text"
      value=""
    />
  </form>
  <h3>
    Death Note
  </h3>
  <h2>
    Cargando...
  </h2>
  <section
    class="card-grid"
  />
</div>
`;
